### ISSUE TITLE
Add support for building wazero with GOARCH=wasm

### ISSUE THREAD

**Issue by supermigo** [tetratelabs/wazero#1526]
../../../../gosrc/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/sysfs/open_file_js.go:11:81: undefined: File
../../../../gosrc/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/sysfs/open_file_js.go:16:12: undefined: O_DIRECTORY
../../../../gosrc/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/sysfs/open_file_js.go:16:26: undefined: O_NOFOLLOW
this error not support wasm

**Comment by achille-roussel** [comment]
Hello @supermigo 

Thanks for reporting tho we're going to need a bit more details from you to be able to help address the issue. 

Would you be able to resubmit the issue and follow the bug reporting template?

**Comment by mathetake** [comment]
I believe this is the error when we try to build downstream user of wazero with GOOS=js GOARCH=wasm: 

```
~/wazero wazevoinit*
$ GOOS=js GOARCH=wasm go build ./...
# github.com/tetratelabs/wazero/internal/sysfs
internal/sysfs/open_file_js.go:11:81: undefined: File
internal/sysfs/open_file_js.go:16:12: undefined: O_DIRECTORY
internal/sysfs/open_file_js.go:16:26: undefined: O_NOFOLLOW
internal/sysfs/sock.go:131:15: undefined: syscall.SHUT_RD
internal/sysfs/sock.go:133:15: undefined: syscall.SHUT_WR
internal/sysfs/sock.go:135:15: undefined: syscall.SHUT_RDWR
internal/sysfs/sock.go:153:28: undefined: syscall.SHUT_RDWR
```

**Comment by mathetake** [comment]
```
GOOS=wasip1 GOARCH=wasm gotip build ./...
# github.com/tetratelabs/wazero/internal/fsapi
internal/fsapi/constants.go:10:24: undefined: syscall.O_DIRECTORY
internal/fsapi/constants.go:11:24: undefined: syscall.O_NOFOLLOW
internal/fsapi/constants.go:12:24: undefined: syscall.O_NONBLOCK
```

Also found that wasip1 doesn't build

**Comment by codefromthecrypt** [comment]
sounds like we are missing an _unsupported.go file somewhere.

**Comment by mathetake** [comment]
seems that this has been resolved together with #1578
